server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectWriteAheadLogs.java

```diff
@@ -80,10 +80,10 @@
 
 public class GarbageCollectWriteAheadLogs {
   private static final Logger log = LoggerFactory.getLogger(GarbageCollectWriteAheadLogs.class);
+  private static final Map<HostAndPort,Long> firstSeenDead = new HashMap<HostAndPort,Long>();
 
   private final AccumuloServerContext context;
   private final VolumeManager fs;
-  private final Map<HostAndPort,Long> firstSeenDead = new HashMap<HostAndPort,Long>();
```

Comment:
+1 to capturing this in an integration test.

You might be able to [use an existing test](https://github.com/apache/accumulo/blob/1.7/test/src/test/java/org/apache/accumulo/test/functional/RestartIT.java) as a starting point. There are test bindings which will make starting/stopping services simple. The tricky point will be identifying a WAL for a tserver before stopping it and then verifying that it does not get deleted after an amount of time greater than the GC's period.